### PR TITLE
ci: pin workdirs on Windows

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
The Bazel cache on Windows includes absolute paths. The normal process
for Azure is to dynamically allocate new top-level folders for each new
build that runs on a given machine. The result of that is that we get
about a one in three chance to get caching for any single Windows build
(it's actually not _quite_ that because we don't run different builds an
equal number of times).

This PR is an attempt at pinning the folder to job mapping by mucking
around in [Azure internals], which may or may not have bad consequences
down the line, assuming it works at all.

[Azure internals]: https://github.com/microsoft/azure-pipelines-agent/blob/master/docs/jobdirectories.md

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

**Comment by garyverhaegen-da** [comment]
Well, that seems to be working as expected. The build did overwrite the "last run" entry for PRs builds, while keeping the other two files intact. And weare working out of `2` despite this being the very first job that runs on this brand new machine.

I don't know how to test this further, but I'll wait for an approving review to deploy this more widely. On approval, the plan is to:
1. Change the `assignment` of `w2` back to `default, and bump the size to 6. Deploy.
2. Disable all `w1` machines in Azure, wait for jobs to drain.
2. Change the size of `w1` to 0. Deploy.
3. Change the `pinning` entry of `w1` to match `w2`. Deploy.
4. Remove pinning as a parameter and inline it instead. This should at this point not result in a new deployment.

File: infra/vsts_agent_windows.tf
Comment (cocreature) [inline infra/vsts_agent_windows.tf:80]: So if I understand this correctly you are trying to get all 3 pipelines to use different directories meaning we still won’t get caching across them but at least each pipeline will cache consistently?
Is there a chance of getting caching across pipelines?

File: infra/vsts_agent_windows.tf
Comment (cocreature) [inline infra/vsts_agent_windows.tf:29]: there are a lot of magic uuids in here. I recommend adding coments that explain how you got them and how to update them when needed.

**Review by cocreature — APPROVED** [review]
Nice! Happy to merge this if we add comments for the magic uuids

File: ci/build.yml
Comment (garyverhaegen-da) [reply ci/build.yml:181]: Yes.

File: infra/vsts_agent_windows.tf
Comment (garyverhaegen-da) [reply infra/vsts_agent_windows.tf:80]: 1. Yes, this is just about pinning each job on the same folder across all machines.
2. I don't know. It may be worth trying to set all the projects to the same folder and see what happens. I'd try that as a separate step, though, as that seems a lot harder to properly test.

File: infra/vsts_agent_windows.tf
Comment (garyverhaegen-da) [reply infra/vsts_agent_windows.tf:29]: To be clear, I did not craft those files: this is all taken literally from the output of one machine (that happened to have run all three jobs) on [this step](https://github.com/digital-asset/daml/pull/12597/files).

I _think_ the UUIDs are the IDs of the jobs in Azure, but I wouldn't recommend trying to change any of this. If we ever get any new job in Azure, I'd still rely on that output and take it as is, rather than try to craft these values.

File: infra/vsts_agent_windows.tf
Comment (garyverhaegen-da) [inline infra/vsts_agent_windows.tf:28]: @cocreature does this work?

Comment (cocreature) [reply infra/vsts_agent_windows.tf:28]: yes, thank you!

**Pull request merged**
